framework/include/utils/MooseTypes.h

```diff
@@ -71,6 +72,21 @@
 #pragma clang diagnostic pop
 #endif
 
+/**
+ * forward declarations
+ */
+namespace libMesh
+{
+template <typename>
+class VectorValue;
+template <typename>
+class TensorValue;
+template <unsigned int, typename>
+class TypeNTensor;
```

Comment:
`#include "libmesh/tensor_value.h"` and `"type_n_tensor.h"` and then you do not need this.